src/defrag.c

```diff
@@ -283,7 +282,7 @@ static void activeDefragZsetNode(void *privdata, void *entry_ref) {
         zskiplistNode *next = x->level[i].forward;
         while (next &&
                (next->score < score ||
-                (next->score == score && sdscmp(zslGetNodeElement(next), ele) < 0))) {
+                (next->score == score && next != node))) {
```

Comment:
If two elements have the same score, they  are sorted lexicographically. That's what the sdscmp < 0 is for. ~~I don't think we can remove it.~~

Maybe we can? This is the defrag code...

Reply:
but in the scope of defrag we are not changing the order right?